[Mantis:27968] Add index to tax_tree.child

Rillke opened the pull request:
This will speed-up queries like

```sql
SELECT
 s.child
FROM
 tax_tree s, tax_tree t
WHERE
 t.child = 8385
AND
 s.lft > t.lft
AND
 s.rgt < t.rgt
AND
 s.tax_tree_id = 18710;
```

in ILIAS instances with a large tax_tree.

-> https://mantis.ilias.de/view.php?id=27968

Performance considerations: While one person usually authors
the question (INSERT/UPDATE), many people in parallel are
accessing the test (SELECT). Many reads, few writes ->
use index.

mjansenDatabay commented:
@Rillke Maybe you could resolve the conflict in `setup/sql/dbupdate_05.php`.

Hi @alex40724 , the TB regularly monitors the list of PRs for old ones. Coud you please merge this (after the conflict is resolved) or request feedback if you have any objections.

Rillke committed:
[Mantis:27968] Add index to tax_tree.child

This will speed-up queries like

SELECT
 s.child
FROM
 tax_tree s, tax_tree t
WHERE
 t.child = 8385
AND
 s.lft > t.lft
AND
 s.rgt < t.rgt
AND
 s.tax_tree_id = 18710;

in ILIAS instances with a large tax_tree.

-> https://mantis.ilias.de/view.php?id=27968

Performance considerations: While one person usually authors
the question (INSERT/UPDATE), many people in parallel are
accessing the test (SELECT). So many reads, few writes ->
use index.

Rillke commented:
@mjansenDatabay @alex40724 Conflict resolved.

alex40724 commented:
Fixed accordingly in 5.4, 6 and trunk

alex40724 commented:
Thanks for the PR.

Rillke commented:
Commits: 70c051a49ed19e05889d3f37074518c340b4e767 357e4d94be7968ad2297080a2e0ab66302419dd3 0ce2df97c38971bbbf888d10de65ff0b77dae4fc
